src/component/tooltip/TooltipHTMLContent.ts

```diff
@@ -212,14 +212,14 @@ function assembleCssText(tooltipModel: Model<TooltipOption>, enableTransition?:
 }
 
 // If not able to make, do not modify the input `out`.
-function makeStyleCoord(out: number[], zr: ZRenderType, appendToBody: boolean, zrX: number, zrY: number) {
+function makeStyleCoord(out: number[], zr: ZRenderType, container: HTMLElement | null, zrX: number, zrY: number) {
     const zrPainter = zr && zr.painter;
 
-    if (appendToBody) {
+    if (container) {
         const zrViewportRoot = zrPainter && zrPainter.getViewportRoot();
         if (zrViewportRoot) {
             // Some APPs might use scale on body, so we support CSS transform here.
-            transformLocalCoord(out, zrViewportRoot, document.body, zrX, zrY);
+            transformLocalCoord(out, zrViewportRoot, container, zrX, zrY);
```

Comment:
I'm also not sure here. As per the context, it seems to be used to handle the case where `document.body` applies the CSS transform. @100pah @Ovilia Could you please provide some information about this?

Reply:
@plainheart  Yes, there is a test case: `test/css-transform.html`

Reply:
the custom contain can also be css transformed. So I think we might need the same logic if the user choose a custom element. That the reason I want to keep container empty so I can know which position logic need to be applied.

Reply:
No problem. You can add some cases to test if the CSS transform also works well for the customized container.